### PR TITLE
Als 4662

### PR DESCRIPTION
**Pull request by Gcolon021**
I continued to get no value for the header even after updating the name to be lowercase. It was due to how `@Context` works. It turns out that `@Context` depends on the classes scope. This caused the Headers to be a singleton and never updated upon each request. I have changed this by passing the headers from the request methods in PicsureRS. The necessary scope for the headers is the `@RequestScope`.

### CODE REVIEW

File: pic-sure-util/src/main/java/edu/harvard/dbmi/avillach/util/Utilities.java [hunk 1]
```diff
@@ -25,8 +27,19 @@ public static HttpClientContext buildHttpClientContext() {
     }
 
     public static String getRequestSourceFromHeader(HttpHeaders headers) {
-        if (headers == null) return "";
-        return Optional.ofNullable(headers.getHeaderString("requestSource")).orElse("");
+        if (headers == null) return "headers are null";
+        return Optional.ofNullable(headers.getHeaderString("request-source")).orElse("request-source header is null");
```
Comment (Luke-Sikina) [inline pic-sure-util/src/main/java/edu/harvard/dbmi/avillach/util/Utilities.java:31]: This is an interesting use of Optional. Usually you see it as a means to both mark return types as nullable and force api consumers to handle nulls. Using an optional purely as a means of control flow is a bit different, but I don't see a reason not to do it. 

You could go a little bit further with this if you're OK with a more general error message:

```suggestion
        return Optional.ofNullable(headers)
            .map(h ->  h.getHeaderString("request-source"))
            .orElse("request-source header is null or headers object is null");
```

File: pic-sure-util/src/main/java/edu/harvard/dbmi/avillach/util/Utilities.java [hunk 2]
```diff
@@ -25,8 +27,19 @@ public static HttpClientContext buildHttpClientContext() {
...
+    }
+
+    public static String convertQueryRequestToString(ObjectMapper mapper, Object searchQueryRequest) {
+        if (mapper == null) return "";
+        if (searchQueryRequest == null) return "";
```
Comment (Luke-Sikina) [inline pic-sure-util/src/main/java/edu/harvard/dbmi/avillach/util/Utilities.java:36]: Bracket, combine predicates

File: pic-sure-util/src/main/java/edu/harvard/dbmi/avillach/util/Utilities.java [hunk 3]
```diff
@@ -25,8 +27,19 @@ public static HttpClientContext buildHttpClientContext() {
...
+        try {
+            return mapper.writeValueAsString(searchQueryRequest);
+        } catch (JsonProcessingException e) {
+            // We don't want to stop code execution if this fails
```
Comment (Luke-Sikina) [inline pic-sure-util/src/main/java/edu/harvard/dbmi/avillach/util/Utilities.java:40]: I think this should be logged

File: pic-sure-util/src/main/java/edu/harvard/dbmi/avillach/util/Utilities.java (hunk 1 above)
Comment (Gcolon021) [reply pic-sure-util/src/main/java/edu/harvard/dbmi/avillach/util/Utilities.java:31]: I prefer the less general error message.  I don't want to have to debug further to determine if it was the header object or the request-source. Additionally, I will convert this from a Optional to a ternary.

File: pic-sure-util/src/main/java/edu/harvard/dbmi/avillach/util/Utilities.java (hunk 3 above)
Comment (Gcolon021) [reply pic-sure-util/src/main/java/edu/harvard/dbmi/avillach/util/Utilities.java:40]: Adding logging.

File: pic-sure-api-war/src/main/java/edu/harvard/dbmi/avillach/service/PicsureInfoService.java [hunk 4]
```diff
@@ -10,21 +11,20 @@
...
 import javax.inject.Inject;
-import javax.ws.rs.core.Context;
 import javax.ws.rs.core.HttpHeaders;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.UUID;
 import java.util.stream.Collectors;
 
+import static edu.harvard.dbmi.avillach.util.Utilities.convertQueryRequestToString;
 import static edu.harvard.dbmi.avillach.util.Utilities.getRequestSourceFromHeader;
```
[the remark this answers is not in the document]
Comment (Gcolon021) [reply pic-sure-api-war/src/main/java/edu/harvard/dbmi/avillach/service/PicsureInfoService.java:21]: I don't disagree, but I do believe that most modern editors display static method as italic text. However, I have changed it to improve clarity.

**Pull request merged**